components/automate-ui/src/app/pages/project/list/project-list.component.ts

```diff
@@ -104,11 +105,9 @@ export class ProjectListComponent implements OnInit, OnDestroy {
       takeUntil(this.isDestroyed),
       // do not update this cache while an update is in progress
       filter(() => !this.applyRulesInProgress)
-    )
-      .subscribe((projectList: Project[]) => {
-        this.statusCache = {};
-        projectList.forEach(p => this.statusCache[p.id] = p.status);
-      });
+    ).subscribe((projectList: Project[]) => {
+      this.statusCache = projectList.reduce((m, p) => m[p.id] = p.status, {});
```

Comment:
```suggestion
      this.statusCache = projectList.reduce((m, p) => { m[p.id] = p.status; return m; }, {});
```

Reply:
Alternately, this feels cleaner to me:
```suggestion
      this.statusCache = Object.assign(...projectList.map(p => ({ [p.id]: p.status })));
```